walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -26,6 +26,19 @@
    * the request ensuring that the attandees don't have any conflicts with other events.
    */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+    if (!request.getOptionalAttendees().isEmpty()) {
+      List<TimeRange> requestAttendeeRanges = filterToRequestAttendeeTimeRangesOptional(
+          events, request.getAttendees(), request.getOptionalAttendees());
+      Collections.sort(requestAttendeeRanges, TimeRange.ORDER_BY_START);
+      requestAttendeeRanges = combineOverlaps(requestAttendeeRanges);
+      List<TimeRange> result =
+          findMeetingRangesWithNoConflict(requestAttendeeRanges, request.getDuration());
+      if (!result.isEmpty()) {
+        return result;
+      } else if (request.getAttendees().isEmpty()) {
+        return Collections.emptyList();
+      }
```

Comment:
This logic should be moved to the top? If no attendees don't need to check any event.

Reply:
I would still have to check for the optional attendees if request.getAttendees() returned an empty collection. 

I added that if statement for the case where there are only optional attendees with no gaps in their schedule. The expected behavior is to return an empty collection. Without this, a collection with an all-day TimeFrame would be returned instead (which is the expected behavior if there are no optional and no mandatory attendees).